### PR TITLE
Fix torch.compile wrong output shape for norm() with negative dim

### LINKED ISSUE #182181: [Inductor] torch.compile changes output shape of x.norm(dim=-1)
### 🐛 Describe the bug

`norm()` behaves differently before and after torch.compile

```python

class TestModule(torch.nn.Module):
    def forward(self, x):
        return x.norm(dim=-1)


m = TestModule()
n = torch.compile(m)

x = torch.rand(8, 3, 1)

print(m(x).shape)
print(n(x).shape) 
```

I get the following output:

```
torch.Size([8, 3])
torch.Size([8, 3, 1]) 
```

### Versions

Vulnerability Tsx async abort:           Mitigation; TSX disabled
Vulnerability Vmscape:                   Mitigation; IBPB before exit to userspace

Versions of relevant libraries:
[pip3] numpy==2.4.4
[pip3] nvidia-cublas==13.1.0.3
[pip3] nvidia-cuda-cupti==13.0.85
[pip3] nvidia-cuda-nvrtc==13.0.88
[pip3] nvidia-cuda-runtime==13.0.96
[pip3] nvidia-cudnn-cu13==9.19.0.56
[pip3] nvidia-cufft==12.0.0.61
[pip3] nvidia-curand==10.4.0.35
[pip3] nvidia-cusolver==12.0.4.66
[pip3] nvidia-cusparse==12.6.3.3
[pip3] nvidia-cusparselt-cu13==0.8.0
[pip3] nvidia-nccl-cu13==2.28.9
[pip3] nvidia-nvjitlink==13.0.88
[pip3] nvidia-nvtx==13.0.85
[pip3] pytorch-lightning==2.6.1
[pip3] torch==2.11.0
[pip3] torchmetrics==1.9.0
[pip3] torchvision==0.26.0
[pip3] triton==3.6.0
[conda] Could not collect

cc @ezyang @gchanan @kadeng @msaroufim @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @muchulee8 @amjames @aakhundov @coconutruben @jataylo

### PR DESCRIPTION
**Pull request by liqiangxl**
The `vector_norm` decomposition has a fast path for when the reduced dimensions all have size 1 (just abs + reshape). This path built the output shape by filtering out dims using `d not in dim`, but `dim` could contain negative indices (e.g. -1) while `enumerate` produces positive indices (0, 1, 2...), so the comparison never matched and the dimension was incorrectly retained in the output.

Canonicalize dim to positive indices before entering the fast path.

Fixes #182181

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

**Review by Lucaskabela — APPROVED** [review]
Thanks for the change - well focused and has test coverage

**Review by Skylion007 — APPROVED** [review]

**Review by karthickai — APPROVED** [review]

**Review by jansel — APPROVED** [review]
